source_code/main_mcu/src/main.c

```diff
@@ -70,10 +70,16 @@ uint32_t* mcu_sp_rh_addresses = 0;
 /* mode (the only mode supported by Cortex M0)                              */
 /****************************************************************************/
 #ifndef EMULATOR_BUILD
+
+extern uint32_t _estack;
+extern uint32_t _sstack;
+
 const uint32_t jump_to_application_function_addr[2] __attribute__((used,section (".flash_start_addr"))) = {HMCRAMC0_ADDR+100,0x200+1};
 void jump_to_application_function(void) __attribute__((used,section (".start_app_function_addr")));
 void jump_to_application_function(void)
 {
+    main_init_stack_tracking();
```

Comment:
I'm afraid that function is only in the flash whenever you flash the solution using your swd programmer.
In the deployed units you may therefore want to add that function here: https://github.com/mooltipass/minible/blob/master/source_code/main_mcu/src/bootloader.c#L62
Get in touch with me so I can explain how you can combine the bootloader solution with the main solution to later flash the result

Reply:
:+1: